Migrate active-directory-plugin documentation from wiki to Github (#100)
jenkinsci/active-directory-plugin: README.md

Active Directory plugin for Jenkins
===================================

[![Build Status](https://ci.jenkins.io/job/Plugins/job/active-directory-plugin/job/master/badge/icon)](https://ci.jenkins.io/job/Plugins/job/active-directory-plugin/job/master/)
[![Jenkins Plugin](https://img.shields.io/jenkins/plugin/v/active-directory.svg)](https://plugins.jenkins.io/active-directory/)
[![GitHub release](https://img.shields.io/github/release/jenkinsci/active-directory.svg?label=changelog)](https://github.com/jenkinsci/active-directory-plugin/releases/latest/)
[![Jenkins Plugin Installs](https://img.shields.io/jenkins/plugin/i/active-directory.svg?color=blue)](https://plugins.jenkins.io/active-directory/)

| Plugin Information                                                                                            |
|---------------------------------------------------------------------------------------------------------------|

> | View Active Directory [on the plugin site](https://plugins.jenkins.io/active-directory) for more information. |

Reviewer: There is a No strong opinion, but there is a badge for it

Author: Removed the plugin information content ass requested by Mark and also resized the images.

> | Plugin Information                                                                                            |
> |---------------------------------------------------------------------------------------------------------------|
> | View Active Directory [on the plugin site](https://plugins.jenkins.io/active-directory) for more information. |
> 
> Older versions of this plugin may not be safe to use. Please review the following warnings before using an older version:
> 
> -   [Improper certificate validation](https://jenkins.io/security/advisory/2019-01-28/#SECURITY-859)
> -   [Man-in-the-middle vulnerability due to missing certificate check](https://jenkins.io/security/advisory/2017-03-20/)

Reviewer 2: No need to link from the plugins site to itself and no need to describe security issues.  They are handled by the plugins site generator.
```suggestion
```

Author: Removed the link to plugin and security

> With this plugin, you can configure Jenkins to authenticate the username and the password through Active Directory.This plugin internally uses two very different implementations, depending on whether Jenkins is running on Windows or non-Windows and if you specify a domain.

Reviewer: ```suggestion
With this plugin, you can configure Jenkins to authenticate the username and the password through Active Directory. This plugin internally uses two very different implementations, depending on whether Jenkins is running on Windows or non-Windows and if you specify a domain.
```

Author: Fixed and pushed this change. Please review.

-   If Jenkins is running on a Windows machine and you do not specify a domain, that machine must be a member of the domain you wish to authenticate against. Jenkins uses ADSI to figure out all the details, so no additional configuration is required.

> -   If Jenkins is running on a non-Windows machine (or you specify one or more domains), then you need to tell Jenkins the name of Active Directory domain(s) to authenticate with. Jenkins then uses DNS SRV records and LDAP service of Active Directory to authenticate users.

Reviewer 2: Minor phrasing change
```suggestion
-   If Jenkins is running on a non-Windows machine (or you specify one or more domains), then you need to tell Jenkins the name of Active Directory domain(s) to use.
Jenkins then uses DNS SRV records and LDAP service of Active Directory to authenticate users.
```

Jenkins recognizes all the groups in Active Directory that the user belongs to, so you can use those to make authorization decisions (for example you can choose the matrix-based security as the authorization strategy and perhaps allow "Domain Admins" to administer Jenkins).

#### Active Directory Health Status

Since the version 2.5 the AD plugin adds a ManagementLink to report a Health Status about the Domain and Domain controllers. In order to correctly use this feature, you should be logged-in into the instance and the cache should be disabled. Then, you will get:

-   The Domain health
    -   DNS resolution
    -   Global Catalog
    -   Ldap Catalog
-   The Domain Controller Health
    -   If the user can login into the DC
    -   The Connection time
    -   The total time in the lookup process

![](docs/images/ad-managementLink-view.png)

#### Fall-back user

> Since the version 2.5 of the AD plugin, you can define a user to fall back in case there is a communication issue between Jenkins and the AD server. On this way, this admin user can be used to continue administering Jenkins in case of communication issues, where usually you were following the link [Disable security](https://wiki.jenkins.io/display/JA/Disable+security). The password of this user is automatically synced with the Jenkins Internal Database by this feature. In order to configure this new feature you should enable \*Use Jenkins Internal Database\* in the AD configuration under Manage Jenkins → Configure Global Security and specify a SINGLE user by its username.

Reviewer: ```suggestion
Since the version 2.5 of the AD plugin, you can define a user to fall back in case there is a communication issue between Jenkins and the AD server.
On this way, this admin user can be used to continue administering Jenkins in case of communication issues, where usually you were following the link [Disable security](https://www.jenkins.io/doc/book/system-administration/security/#disabling-security).
The password of this user is automatically synced with the Jenkins Internal Database by this feature. In order to configure this new feature you should enable \*Use Jenkins Internal Database\* in the AD configuration under Manage Jenkins → Configure Global Security and specify a SINGLE user by its username.
```

Fixed the link + line breaks for easier reviews

Author: Fixed and pushed this change.

  
![](docs/images/ad-internalJenkinsUser.png)

> #### [SECURITY-251](https://wiki.jenkins.io/display/JENKINS/SECURITY-251) Active Directory Plugin did not verify certificate of AD server

Reviewer: ```suggestion
#### [SECURITY-251](https://wiki.jenkins.io/display/JENKINS/SECURITY-251) Active Directory Plugin did not verify certificate of AD server
```

I believe this page has been already moved to jenkins.io. CC @daniel-beck

Author: Have changed the link to "https://www.jenkins.io/security/advisory/2017-03-20/#SECURITY-251". Not sure if this is the correct link.

#### If using Domain controller check that all servers on the farm are working correctly

In case, we are using a Domain Controller like in the example below we might want to list all the AD servers in the farm by using:

    nslookup <DOMAIN_CONTROLLER>

![](docs/images/ad-domain-controller-configuration.png){width="600"}

It might happen that one of the servers in the farm is incorrectly replicated and the ad-plugin is sticky with this one, so we might want to check with ldapsearch command or the Test button in the GUI that all the servers are working correctly trying to look for an user on the tree.

#### If using Domain controller check that all servers on the farm are working correctly

You can check this by using:

    nslookup -q=SRV _ldap._tcp.<DOMAIN_NAME>

    nslookup -q=SRV _gc._tcp.<DOMAIN_NAME>

## Warning for 1.37

Be careful if you intend to install version 1.37. It has been known to cause excessive load on Active Directory authentication servers. If you install this version you should carefully monitor traffic on relevant ports, e.g.: `tcpdump port 389 or 3268`.

> ## Changelog

Reviewer: Please move the changelog to a separate CHANGELOG.md file
